find_todo_fixme.sh

```
@@ -0,0 +1,83 @@
+  PATHS=()
+  cd "$SINGLE_DIR_ROOT" || exit
+fi
+
+# Get the search pattern from the environment variable or use a default
+SEARCH_PATTERN=${FIND_TODO_FIXME_SEARCH_PATTERN:-'(TODO|FIXME|HACK|FIX):\s'}
+
+# Set up the ripgrep command
+RG_PREFIX=(rg
+    --column
+    --hidden
+    $(array_join ${USE_GITIGNORE_OPT+"${USE_GITIGNORE_OPT[@]}"})
```

Shellcheck

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)